src/configuration.js

```diff
@@ -1,8 +1,11 @@
+const cosmiconfig = require('cosmiconfig');
+import { readSync, readFileSync } from 'fs';
+import getGraphQLProjectConfig from 'graphql-config';
+import path from 'path';
+
 import defaultRules from './rules/index.js';
 import JSONFormatter from './formatters/json_formatter.js';
 import TextFormatter from './formatters/text_formatter.js';
-import getGraphQLProjectConfig from 'graphql-config';
-import { readSync, readFileSync } from 'fs';
 
 export class Configuration {
   constructor(options, stdinFd) {
```

Comment:
I guess that's the disadvantage of not using something like TypeScript. Maybe in the future we could port it over.

One thing to keep in mind is that `except` and `only` are both arrays at this point since the transformation is done here: https://github.com/goldcaddy77/graphql-schema-linter/blob/68a5c6916328a82c19553e2677d40a771240dc79/src/runner.js#L8

Maybe we should also :fire: `args` and have `schemaPath` that way the `runner.js` does the `args` handling.

I wonder if `searchDirectory` belongs in this list though. All the other options are _actual_ options. I kinda feel like it should be `constructor(options, stdinFd, searchDirectory)`.

That way `options` only contains the things you can actually specify via CLI options or configuration file.